jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/Layers.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.buildfile;
+
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.api.buildplan.FileEntry;
+import com.google.cloud.tools.jib.api.buildplan.LayerObject;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/** Class to convert between different layer representations. */
+class Layers {
+
+  /**
+   * Convert a layer spec to a list of layer objects.
+   *
+   * <p>Does not handle missing directories for files added via this method. We can either prefill
+   * directories here, or allow passing of the file entry information directly to the reproducible
+   * layer builder
+   *
+   * @param buildRoot the directory to resolve relative paths, usually the directory where the build
+   *     config file is located
+   * @param layersSpec a layersSpec containing configuration for all layers
+   * @return a {@link List} of {@link LayerObject} to use as part of a buildplan
+   * @throws IOException if traversing a directory fails
+   */
+  static List<LayerObject> toLayers(Path buildRoot, LayersSpec layersSpec) throws IOException {
```

Comment:
For my understanding, this method will be expanded to support other layers in addition to FileLayer in the future?

Reply:
Correct, but I'm not sure what needs to be updated here?

Reply:
Nice, might be worth adding a line to the PR summary or a TODO at beginning of the method that mentions that the change is for FileLayer.

Reply:
Added to PR description